Regression: `wasm-unstable-single-threaded` required for sync-only wasm32 crates in 0.30

In 0.30, any crate using uniffi on wasm32 fails to compile without `wasm-unstable-single-threaded`, even when it has no async code. In 0.29 the feature was only needed for async wasm32 code.

### STR

Take any sync-only uniffi crate (e.g. the [arithmetic example](https://github.com/mozilla/uniffi-rs/tree/main/examples/arithmetic)) and build it for wasm32 without the feature:

```sh
cargo build --target wasm32-unknown-unknown
```

**Expected:** compiles (as in 0.29)

**Actual:**
```
error[E0277]: `F` cannot be sent between threads safely
  --> uniffi_core-0.30.0/src/ffi/rustfuture/mod.rs:96:42
96 | impl<T, F> UniffiCompatibleFuture<T> for F where F: Future<Output = T> {}
   |                                          ^ `F` cannot be sent between threads safely
note: required by a bound in `rustfuture::UniffiCompatibleFuture`
43 | pub trait UniffiCompatibleFuture<T>: Future<Output = T> + Send {}
```

--- a/uniffi_core/src/ffi/rustfuture/mod.rs
+++ b/uniffi_core/src/ffi/rustfuture/mod.rs
@@ -47,12 +47,15 @@ pub trait FutureLowerReturn<UT>: LowerReturn<UT> {}
 
 /// The `Send` bound is required because the Foreign code may call the
 /// `rust_future_*` methods from different threads.
-#[cfg(not(target_arch = "wasm32"))]
+#[cfg(not(all(target_arch = "wasm32", feature = "wasm-unstable-single-threaded")))]
 impl<T, F> UniffiCompatibleFuture<T> for F where F: Future<Output = T> + Send {}
 #[cfg(not(all(target_arch = "wasm32", feature = "wasm-unstable-single-threaded")))]
 impl<UT, LR> FutureLowerReturn<UT> for LR where LR: LowerReturn<UT> + Send {}
 
-/// `Future`'s on WASM32 are not `Send` because it's a single threaded environment.
+/// `Future`'s on WASM32 are typically not `Send` because it's a single threaded environment.
+///
+/// Users can opt into allowing non-Send futures by using the `wasm-unstable-single-threaded`
+/// feature.  This creates a `UniffiCompatibleFuture` impl for non-Send futures.
 ///
 /// # Safety:
 ///
@@ -91,8 +94,7 @@ impl<UT, LR> FutureLowerReturn<UT> for LR where LR: LowerReturn<UT> + Send {}
 /// [transferable]: (https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Transferable_objects
 #[cfg(all(target_arch = "wasm32", feature = "wasm-unstable-single-threaded"))]
 pub trait UniffiCompatibleFuture<T>: Future<Output = T> {}
-
-#[cfg(target_arch = "wasm32")]
+#[cfg(all(target_arch = "wasm32", feature = "wasm-unstable-single-threaded"))]
 impl<T, F> UniffiCompatibleFuture<T> for F where F: Future<Output = T> {}
 #[cfg(all(target_arch = "wasm32", feature = "wasm-unstable-single-threaded"))]
 impl<UT, LR> FutureLowerReturn<UT> for LR where LR: LowerReturn<UT> {}

